x-pack/plugin/eql/src/main/java/org/elasticsearch/xpack/eql/execution/sequence/TimeOrdinal.java

```diff
@@ -0,0 +1,60 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+package org.elasticsearch.xpack.eql.execution.sequence;
+
+import org.elasticsearch.xpack.ql.capabilities.Resolvable;
+import org.elasticsearch.xpack.ql.expression.Attribute;
+
+import java.util.List;
+import java.util.Objects;
+
+import static java.util.Arrays.asList;
+import static java.util.Collections.singletonList;
+
+/**
+ * Time ordinal for a given event.
+ * It is an internal construct that wraps the mandatory timestamp attribute and the optional application tie-breaker.
+ */
+public class TimeOrdinal implements Resolvable {
```

Comment:
`Ordinal` doesn't feel right in this context tbh. How about `TimeOrdering`?

Reply:
It's not the order per se but rather where in the stream an event stands. And it's not just the time but also the tiebreaker that define the position in the stream.